server/src/main/java/org/opensearch/index/shard/ShardStateMetadata.java

```diff
@@ -56,17 +56,35 @@ public final class ShardStateMetadata {
     private static final String PRIMARY_KEY = "primary";
     private static final String INDEX_UUID_KEY = "index_uuid";
     private static final String ALLOCATION_ID_KEY = "allocation_id";
+    private static final String INDEX_DATA_LOCATION_KEY = "index_data_location";
+
+    /**
+     * Enumeration of types of data locations for an index
+     */
+    public enum IndexDataLocation {
+        /**
+         * Indicates index data is on the local disk
+         */
+        LOCAL,
+        /**
+         * Indicates index data is remote, such as for a searchable snapshot
+         * index
+         */
+        REMOTE
+    }
 
     public final String indexUUID;
     public final boolean primary;
     @Nullable
     public final AllocationId allocationId; // can be null if we read from legacy format (see fromXContent and MultiDataPathUpgrader)
+    public final IndexDataLocation indexDataLocation;
 
-    public ShardStateMetadata(boolean primary, String indexUUID, AllocationId allocationId) {
+    public ShardStateMetadata(boolean primary, String indexUUID, AllocationId allocationId, IndexDataLocation indexDataLocation) {
```

Comment:
nit - overload this with the default IndexDataLocation  as LOCAL?

Reply:
Sure